From ad7ab3a42ad85adfa74ce6cae8c2c8658c6ece89 Mon Sep 17 00:00:00 2001
From: Vanessa Maegima <vanessa.maegima@foundries.io>
Date: Fri, 10 Nov 2023 10:30:16 -0300
Subject: [PATCH 07/24] arch: arm: dts: portenta-x8: fix i2c booting errors

This fixes error:

    U-Boot SPL 2022.04+fio+gd7db7d85cda (Jul 12 2023 - 15:19:02 +0000)
    power_bd71837_init
    wait_for_sr_state: failed sr=81 cr=a0 state=2020
    i2c_init_transfer: failed for chip 0x4b retry=0
    wait_for_sr_state: failed sr=81 cr=a0 state=2020
    i2c_init_transfer: failed for chip 0x4b retry=1
    wait_for_sr_state: failed sr=81 cr=a0 state=2020
    i2c_init_transfer: failed for chip 0x4b retry=2
    i2c_init_transfer: give up i2c_regs=0x30a30000
    Can't find PMIC:BD71837

Signed-off-by: Vanessa Maegima <vanessa.maegima@foundries.io>
---
 arch/arm/dts/portenta-x8-u-boot.dtsi | 36 ++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm/dts/portenta-x8-u-boot.dtsi b/arch/arm/dts/portenta-x8-u-boot.dtsi
index 90b27823e75..9a2de717830 100644
--- a/arch/arm/dts/portenta-x8-u-boot.dtsi
+++ b/arch/arm/dts/portenta-x8-u-boot.dtsi
@@ -100,14 +100,50 @@
 	u-boot,dm-spl;
 };
 
+&i2c2 {
+	u-boot,dm-spl;
+};
+
+&{/soc@0/bus@30800000/i2c@30a30000/pmic@4b} {
+	u-boot,dm-spl;
+};
+
+&{/soc@0/bus@30800000/i2c@30a30000/pmic@4b/regulators} {
+	u-boot,dm-spl;
+};
+
 &pinctrl_i2c1 {
 	u-boot,dm-spl;
 };
 
+&pinctrl_i2c2 {
+	u-boot,dm-spl;
+};
+
 &pinctrl_pmic {
 	u-boot,dm-spl;
 };
 
+&gpio1 {
+	u-boot,dm-spl;
+};
+
+&gpio2 {
+	u-boot,dm-spl;
+};
+
+&gpio3 {
+	u-boot,dm-spl;
+};
+
+&gpio4 {
+	u-boot,dm-spl;
+};
+
+&gpio5 {
+	u-boot,dm-spl;
+};
+
 &fec1 {
 	phy-reset-gpios = <&gpio3 6 GPIO_ACTIVE_LOW>;
 	phy-reset-duration = <100>;
-- 
2.34.1

